bridges/snowbridge/primitives/inbound-queue/src/v2/converter.rs

```diff
@@ -0,0 +1,781 @@
+		InboundQueueLocation,
+		ConvertAssetId,
+		GatewayProxyAddress,
+		EthereumUniversalLocation,
+		GlobalAssetHubLocation,
+	> ConvertMessage
+	for MessageToXcm<
+		CreateAssetCall,
+		CreateAssetDeposit,
+		EthereumNetwork,
+		InboundQueueLocation,
+		ConvertAssetId,
+		GatewayProxyAddress,
+		EthereumUniversalLocation,
+		GlobalAssetHubLocation,
+	>
+where
+	CreateAssetCall: Get<CallIndex>,
+	CreateAssetDeposit: Get<u128>,
+	EthereumNetwork: Get<NetworkId>,
+	InboundQueueLocation: Get<InteriorLocation>,
+	ConvertAssetId: MaybeEquivalence<TokenId, Location>,
+	GatewayProxyAddress: Get<H160>,
+	EthereumUniversalLocation: Get<InteriorLocation>,
+	GlobalAssetHubLocation: Get<Location>,
+{
+	fn convert(message: Message) -> Result<Xcm<()>, ConvertMessageError> {
+		let message = Self::prepare(message)?;
+
+		log::trace!(target: LOG_TARGET,"prepared message: {:?}", message);
+
+		let network = EthereumNetwork::get();
+
+		let mut instructions = vec![
+			DescendOrigin(InboundQueueLocation::get()),
+			UniversalOrigin(GlobalConsensus(network)),
+			ReserveAssetDeposited(message.execution_fee.clone().into()),
+			PayFees { asset: message.execution_fee.clone() },
+		];
+
+		// Make the origin account on AH the default claimer. This account can transact
+		// on AH once it gets full EVM support.
+		let default_claimer = Location::new(
+			0,
+			[AccountKey20 {
+				// Set network to `None` to support future Plaza EVM chainid by default.
+				network: None,
+				// Ethereum account ID
+				key: message.origin.as_fixed_bytes().clone(),
+			}],
+		);
+
+		let claimer = message.claimer.unwrap_or(default_claimer);
+
+		instructions.push(SetHints {
+			hints: vec![AssetClaimer { location: claimer.clone() }]
+				.try_into()
+				.expect("checked statically, qed"),
+		});
+
+		let mut reserve_deposit_assets = vec![];
+		let mut reserve_withdraw_assets = vec![];
+
+		for asset in message.assets {
+			match asset {
+				AssetTransfer::ReserveDeposit(asset) => reserve_deposit_assets.push(asset),
+				AssetTransfer::ReserveWithdraw(asset) => reserve_withdraw_assets.push(asset),
+			};
+		}
+
+		instructions.push(ReserveAssetDeposited(reserve_deposit_assets.into()));
+		instructions.push(WithdrawAsset(reserve_withdraw_assets.into()));
+
+		// If the message origin is not the gateway proxy contract, set the origin to
+		// the original sender on Ethereum. Important to be before the arbitrary XCM that is
+		// appended to the message on the next line.
+		if message.origin != GatewayProxyAddress::get() {
+			instructions.push(DescendOrigin(
+				AccountKey20 { key: message.origin.into(), network: None }.into(),
+			));
+		}
+
+		// Add the XCM sent in the message to the end of the xcm instruction
+		instructions.extend(message.remote_xcm.0);
+		instructions.push(RefundSurplus);
+		// Refund excess fees to the claimer
+		instructions.push(DepositAsset {
+			assets: Wild(AllOf { id: message.execution_fee.id, fun: WildFungible }),
+			beneficiary: claimer,
+		});
```

Comment:
I'd suggest removing the above 2 instructions, as the default claimer won't work before AH support H160 address.

IMO, the refund/claimer can be part of the `message.remote_xcm` provided by the end user.  I'd assume the extra `execution_fee` is tiny, so it's not unacceptable even if it gets trapped, especially compared to the entire xcm failing due to forgetting to set the claimer.

Reply:
Another option as we discussed is to use snowbridge sovereign as the default claimer, @vgeddes @claravanstaden  which one do you prefer?

Reply:
Yes, I still need to add it - it slipped my mind.

Reply:
Addressed in https://github.com/paritytech/polkadot-sdk/pull/7402/commits/a0f5e4b96cc7afaaa29d5c8f027579de5921f857.

Reply:
I agree with Ron, I think removing the hardcoded handling of excess fees is a good idea.

It is very easy for the UX layers to include those instructions in the XCM passed to the bridge.

It also simplifies the handling of the xcm topic as discussed in  https://github.com/vgeddes/polkadot-sdk/pull/32#discussion_r1983681377